lib/panel-league/garbage.js

```diff
@@ -143,9 +144,90 @@ const releaseGarbage = ((state) => {
   slabsToRemove.forEach((slab) => garbage.splice(garbage.indexOf(slab), 1));
 });
 
+
+const comboToSlabs = ((args) => {
+  const {minWidth, state} = args;
+  let slabs = [];
+  let size = args.size;
+  let x = 0;
+  if (size < minWidth) {
+    size = minWidth;
+  }
+  if (size > state.width) {
+    let overflow = (size % state.width);
+    if (overflow === 0) {
+      overflow = state.width;
+    }
+    const breakSize = Math.floor(overflow / 2) + minWidth;
+    const extraSlabs = comboToSlabs({
+      size: size - breakSize,
+      minWidth,
+      state
+    });
+    slabs = slabs.concat(extraSlabs);
+    size = breakSize;
+  }
+  if (size === minWidth) {
+    let right = 0;
+    let left = 0;
+    state.garbage.forEach((slab) => {
+      if (slab.width === minWidth) {
+        if (slab.x === 0) {
+          ++left;
+        } else {
+          ++right;
+        }
+      }
+    });
+    if (left > right) {
+      x = minWidth;
+    }
+  } else if (size < state.width) {
+    const RNG = JKISS31.unserialize(state.RNG);
+    x = (RNG.step() % (state.width - size));
+    state.RNG = RNG.serialize();
+  }
+  slabs.push({
+    x,
+    uuid: uuidV4(),
+    width: size,
+    height: 1
+  });
+  return slabs;
+});
+
+const effectToSlabs = ((effect, state) => {
+  switch (effect.type) {
+    case 'matchMade':
+    case 'chainMatchMade':
+      if (effect.indices.length > 3) {
+        return comboToSlabs({
+          size: effect.indices.length - 1,
+          minWidth: 3,
+          state,
+        });
+      }
+      break;
+    case 'chainDone':
+      if (effect.chainNumber) {
+        return [{
+          x: 0,
+          uuid: uuidV4(),
+          width: state.width,
+          height: effect.chainNumber,
+        }]
+      }
+      break;
+    default:
+      break;
+  }
+  return [];
```

Comment:
I don't follow the intuition. The function tells what kind of garbage results from a game effect. It's a list because combo effects turn in to multiple garbage slabs.